packages/page-staking/src/Suspensions/Suspensions.tsx

```diff
@@ -0,0 +1,118 @@
+// Copyright 2017-2022 @polkadot/app-staking authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect, useMemo, useState } from 'react';
+
+import useErasStartSessionIndexLookup from '@polkadot/app-staking/Performance/useErasStartSessionIndexLookup';
+import { SuspensionEvent } from '@polkadot/app-staking/Suspensions/index';
+import { createNamedHook, useApi } from '@polkadot/react-hooks';
+import { Vec } from '@polkadot/types';
+import { EventRecord, Hash } from '@polkadot/types/interfaces';
+import { Codec } from '@polkadot/types/types';
+
+type SuspensionReasons = [string, string, number][];
+
+function parseEvents (events: EventRecord[]): SuspensionReasons {
+  return events.filter(({ event }) => {
+    return event.section === 'elections' && event.method === 'BanValidators';
+  }).map(({ event }) => {
+    const raw = event.data[0] as unknown as Codec[][];
+
+    const reasons: SuspensionReasons = raw.map((value) => {
+      const account = value[0].toString();
+      const reasonAndEra = value[1].toHuman() as unknown as Record<string, Codec>;
+
+      const reasonTypeAndValue = reasonAndEra.reason as unknown as Record<string, string>;
+      const reasonType = Object.keys(reasonTypeAndValue)[0];
+      const reasonValue = Object.values(reasonTypeAndValue)[0];
+      const era = Number(reasonAndEra.start.toString());
+
+      if (reasonType === 'OtherReason') {
+        return [account, reasonValue, era];
+      } else if (reasonType === 'InsufficientUptime') {
+        return [account, 'Insufficient uptime in at least ' + reasonValue + ' sessions', era];
+      } else {
+        return [account, reasonType + ': ' + reasonValue, era];
+      }
```

Comment:
Since you return from every clause you shouldn't need the 	`return` keyword?

Reply:
I think as long as I use `{}` in `map` , I need to specify explicitly `return`